### PR TITLE
system tests: authfile-exists: minor cleanup

### PR DESCRIPTION
**Pull request by edsantiago**
Quick followup to #19348:
 - refactor into table form, for legibility
 - add tests for 'podman kube play' and 'podman run'
 - slightly cleaner message on failure

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

File: test/system/030-run.bats [hunk 1]
```diff
@@ -1158,35 +1158,46 @@ EOF
     run_podman rm -f -t0 $ctr
 }
 
-@test "podman --authfile=/tmp/bogus " {
+@test "podman --authfile=nonexistent-path" {
+    # List of commands to be tested. These all share a common authfile check.
+    #
+    # Table format is:
+    #   podman command | arguments | 'x' if it does not work with podman-remote
```
Comment (vrothberg) [inline test/system/030-run.bats:1165]: Non-blocking nit
```suggestion
    #   podman command | arguments | '-' if it does not work with podman-remote
```

**Review by vrothberg — APPROVED** [review]
LGTM

File: test/system/030-run.bats (hunk 1 above)
Comment (edsantiago) [reply test/system/030-run.bats]: duh. Thank you.

**Comment by edsantiago** [comment]
IMO the diff is impossible to review, I recommend just jumping straight to the code: https://github.com/containers/podman/blob/472388157f921a10aec8c34120759499db3dfa84/test/system/030-run.bats#L1161

**Comment by edsantiago** [comment]
Thanks. `parse_table` is a pain to use, it always takes me a long time to write each one, and I always have to grep for other examples. High initial cost, but (I hope) much lower maintenance cost.

**Pull request merged**